login_password_page.php

```diff
@@ -81,8 +81,13 @@
 
 # Get the user id and based on the user decide whether to continue with native password credential
 # page or one provided by a plugin.
+#
+# OR
+#
+# if the plugin provides autoprovisioning, than we can also redirect to the plugin page
+#
 $t_user_id = auth_get_user_id_from_login_name( $t_username );
-if( $t_user_id !== false && auth_credential_page( '', $t_user_id ) != AUTH_PAGE_CREDENTIAL ) {
+if( ( $t_user_id !== false || authenticator_autoprov_capable() ) && auth_credential_page( '', $t_user_id ) != AUTH_PAGE_CREDENTIAL ) {
```

Comment:
Should we have a logic like the following before this block:

```
if( $t_user_id === false ) {
    $t_user_id = auth_provision_user( $p_username );
}

// then use the existing logic
```

In this case, we always give plugins the change to provision the user, but they make decide that a username doesn't exist in the identity system and hence it shouldn't be provisioned or they don't support auto-provision.  Hence, we replace `authenticator_autoprov_capable()` with `auth_provision_user( $p_username )`.  The auth_provision_user() returns user id created or false.

Reply:
I think this will not work with SSO systems. I mean, if you set $credentials_page to support SSO/external password aqusition page, than you'll not get redirected to the plugin page.